pip/req/req_file.py

```diff
@@ -97,166 +70,152 @@ def parse_requirements(filename, finder=None, comes_from=None, options=None,
-            if opt == '-i' or opt == '--index-url':
-                finder.index_urls = [value]
-            elif opt == '--extra-index-url':
-                finder.index_urls.append(value)
-            elif opt == '--allow-external':
-                finder.allow_external |= set([normalize_name(value).lower()])
-            elif opt == '--allow-insecure':
-                # Remove after 7.0
-                finder.allow_unverified |= set([normalize_name(line).lower()])
-            elif opt == '-f' or opt == '--find-links':
-                # FIXME: it would be nice to keep track of the source
-                # of the find_links: support a find-links local path
-                # relative to a requirements file.
-                req_dir = os.path.dirname(os.path.abspath(filename))
-                relative_to_reqs_file = os.path.join(req_dir, value)
-                if os.path.exists(relative_to_reqs_file):
-                    value = relative_to_reqs_file
-                finder.find_links.append(value)
-
-        # ---------------------------------------------------------------------
-        elif linetype == IGNORE:
-            pass
-
-
-def parse_line(line):
-    if not line.startswith('-'):
-        # Split the requirement from the options.
-        if ' --' in line:
-            req, opts = line.split(' --', 1)
-            opts = parse_requirement_options('--%s' % opts)
-        else:
-            req = line
-            opts = {}
-
-        return REQUIREMENT, (req, opts)
-
-    firstword, rest = partition_line(line)
-    # -------------------------------------------------------------------------
-    if firstword == '-e' or firstword == '--editable':
-        return REQUIREMENT_EDITABLE, rest
-
-    # -------------------------------------------------------------------------
-    if firstword == '-r' or firstword == '--requirement':
-        return REQUIREMENT_FILE, rest
-
-    # -------------------------------------------------------------------------
-    if firstword in parser_flags:
-        if rest:
-            msg = 'Option %r does not accept values.' % firstword
-            raise RequirementsFileParseError(msg)
-        return FLAG, firstword
+def process_line(line, filename, line_number, finder=None, comes_from=None,
+                 options=None, session=None, cache_root=None):
+    """
+    Process a single requirements line; This can result in creating/yielding
+    requirements, or updating the finder.
+    """
 
-    # -------------------------------------------------------------------------
-    if firstword in parser_options:
-        if not rest:
-            msg = 'Option %r requires value.' % firstword
-            raise RequirementsFileParseError(msg)
-        return OPTION, (firstword, rest)
+    parser = build_parser()
+    args = shlex.split(line)
+    opts, args = parser.parse_args(args)
+    req = None
+
+    if args:
+        # don't allow multiple requirements
+        if len(args) > 1:
```

Comment:
good catch.   I think I'll remove  the ">1" check all together, and just pass all the args (as a joined string) into the InstallRequirement constructor.   If they did really try to put multiple requirements on one line, it will fail at that point.   that's good enough I think.

and I'll certainly add a test for this.